tests/jetty-test-common/src/main/java/org/eclipse/jetty/tests/OpenIdProvider.java

```
@@ -199,38 +211,37 @@ protected void doGetAuthEndpoint(Request request, Response response, Callback ca
     {
         Fields parameters = Request.getParameters(request);
 
-        if (!clientId.equals(parameters.getValue("client_id")))
+        String redirectUri = parameters.getValue("redirect_uri");
+        if (!redirectUris.contains(redirectUri))
         {
-            Response.writeError(request, response, callback, HttpStatus.FORBIDDEN_403, "invalid client_id");
+            Response.writeError(request, response, callback, HttpStatus.FORBIDDEN_403, "invalid redirect_uri");
             return;
         }
 
-        String redirectUri = parameters.getValue("redirect_uri");
-        if (!redirectUris.contains(redirectUri))
+        String state = parameters.getValue("state");
+        if (state == null)
         {
-            LOG.warn("invalid redirectUri {}", redirectUri);
-            Response.writeError(request, response, callback, HttpStatus.FORBIDDEN_403, "invalid redirect_uri");
+            Response.writeError(request, response, callback, HttpStatus.FORBIDDEN_403, "no state param");
             return;
         }
 
-        String scopeString = parameters.getValue("scope");
-        List<String> scopes = (scopeString == null) ? Collections.emptyList() : Arrays.asList(scopeString.split(" "));
-        if (!scopes.contains("openid"))
+        if (!clientId.equals(parameters.getValue("client_id")))
         {
-            Response.writeError(request, response, callback, HttpStatus.FORBIDDEN_403, "no openid scope");
+            sendError(response, callback, redirectUri, state, "invalid_request_object", "invalid client_id", "example.com/client_id");
```

`example.com`?
Is this mandated by OpenID, or a leftover?

---

This `OpenIdProvider` is just a testing utility, so that we can test our implementation locally without relying on an external identity provider.

So `example.com/client_id` is just a dummy value representing the error_uri field.